### PR TITLE
GMDX-383 partial, refactor token store

### PR DESCRIPTION
**Pull request by bjdupuis**
Refactor the token store. The refactoring of the MobileMessenger will include the ability to pass in a TokenStore. Note that there will be additional changes that make this change make more sense... we don't instantiate the DefaultTokenStore in the MobileMessenger classes as it's done right now moving forward, that's just to get this refactor in a PRable state.

### CODE REVIEW

File: transport/src/androidMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.genesys.cloud.messenger.transport.util
+
+import java.util.*
```
Comment (AfanasievAnton) [inline transport/src/androidMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenGenerator.kt:3]: lint will complain on the wildcard import.

File: transport/src/androidMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenGenerator.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+actual object TokenGenerator {
+    actual fun generate() = UUID.randomUUID().toString()
+}
```
Comment (AfanasievAnton) [inline transport/src/androidMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenGenerator.kt:7]: Lint will also complain on missing new line in the end of the file.
you can run the `./gradlew lintKotlin`  command to verify.

**Review by AfanasievAnton — APPROVED** [review]

File: transport/src/androidMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenGenerator.kt (hunk 2 above)
Comment (bjdupuis) [reply transport/src/androidMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenGenerator.kt]: Thanks for the `lintKotlin` tip. Fixed.

File: transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenGenerator.kt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package com.genesys.cloud.messenger.transport.util
+
+expect object TokenGenerator {
+    fun generate(): String
```
Comment (crumpf) [inline transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenGenerator.kt:4]: I see the `actual` implementations provide a platform-specific UUID and I noticed that we already have a way to generate platform-specific UUID strings in Platform::randomUUID().

Comment (bjdupuis) [reply transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenGenerator.kt]: Used it!

**Review by crumpf — APPROVED** [review]

**Pull request merged**